modules/volunteers/admin/sitesupport/processReturn.js

```diff
@@ -0,0 +1,89 @@
+/* jshint esversion: 6 */
+/* globals  apiRequest, showSidebar, confirmbox */
+
+export default {
+  data() {
+    return {
+      claims: null,
+      creditHours: 0,
+    }
+  },
+  methods: {
+    processReturn() {
+      confirmbox('Confirm Gift Return',
+        'Are the returned gifts correct?').then(() => {
+        var deleted = false;
+        this.claims.forEach((c) => {
+          if (c.returned) {
+            apiRequest('DELETE','/volunteer/claims/' + c.id)
+              .then(() => {
+                if (!deleted) {
+                  document.getElementById('return_success_dlg').style.display = 'block';
+                }
+                deleted = true;
+              });
+          }
+        });
+      });
+    },
+    markDelete(index, claim) {
+      claim.returned = !claim.returned;
+      var total = 0;
+      this.claims.forEach((c) => {
+        if (c.returned && c.reward.promo != '1') {
+          total += parseFloat(c.reward.value);
+        }
+      });
+
+      this.creditHours = Math.round(total * 100) / 100;
+    },
+    show(claims) {
+      this.claims = JSON.parse(JSON.stringify(claims));
```

Comment:
Similar question, can you help me understand the need to deep copy here?

Reply:
The deep copy is required so that we do not edit the record passed to us from the table itself. Otherwise as we change values the change live on that record and we would require robust undo logic to resort the state if the edit sidebar is closed or canceled.